### PR TITLE
Support opening files with any extension

### LINKED ISSUE #6: Allow any file extension for Open with...
Since HDF5 is designed to be embedded into other file formats, HDF5 files may have arbitrary extensions. Thus it would be nice if the "Open with" context menu could be used to open any file.

### PR DESCRIPTION
**Pull request by axelboc**
Fix #6 

I think I've finally found a way to allow opening files with any extension. I've added a new [custom editor contribution point](https://code.visualstudio.com/api/references/contribution-points#contributes.customEditors) with a `priority` set to `option`. This allows showing H5Web in the "Open with..." menu of every file as suggested in #6, while still not making it the "default" editor for every file (which would break VS Code, obviously).

The downside is that H5Web now appears twice in the "Open with" menu of files with supported extensions. For this reason, I've named the fallback editor "H5Web (any extension)".

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,16 @@
...
+      },
+      {
+        "viewType": "h5web.fallback-viewer",
+        "displayName": "H5Web (any extension)",
+        "selector": [
+          {
+            "filenamePattern": "*.*"
+          }
+        ],
+        "priority": "option"
```
[the remark this answers is not in the document]
Comment (loichuder) [reply package.json:54]: Yeah, I wouldn't worry too much :wink:

**Review by loichuder — APPROVED** [review]
Nice ! It makes indeed sense to have a default editor for surefire extensions and an optional editor for the rest.

**Comment by mkitti** [comment]
It may be useful to take inspiration from https://github.com/microsoft/vscode-hexeditor

**Comment by axelboc** [comment]
@mkitti anything specific in mind? The hex editor is [not associated with _any_ specific extension](https://github.com/microsoft/vscode-hexeditor/blob/main/package.json#L86-L97), but in our case, we do want the H5Web editor to be the default editor for at least _some_ extensions. If I've missed something, please feel free to open a PR! :pray:

**Comment by mkitti** [comment]
The `workbench.editorAssociations` they mention in the README seems like something to look into

```
"workbench.editorAssociations": {
    "*.hex": "hexEditor.hexedit",
    "*.ini": "hexEditor.hexedit"
},
```

**Comment by axelboc** [comment]
Nice, I can definitely add this to the README for sure, good call. I can't quite picture whether it's doable for the extension to automatically associate itself with the default extensions by updating this setting. :shrug:

**Pull request merged**